libs/wallet/src/web3-react/hooks/useIsSmartContractWallet.ts

```diff
@@ -0,0 +1,87 @@
+import { useState, useEffect, useMemo } from 'react'
+
+import { useWeb3React } from '@web3-react/core'
+
+import useSWR from 'swr'
+import { useAsyncMemo } from 'use-async-memo'
+import { useWalletInfo } from '../../api/hooks'
+import { useWalletMetaData } from './useWalletMetadata'
+import { getIsAmbireWallet } from '../../api/utils/connection'
+import { Contract } from '@ethersproject/contracts'
+import { getProviderOrSigner } from '@cowswap/common-utils'
+import { ARGENT_WALLET_DETECTOR_ADDRESS } from '@cowswap/common-const'
+import { ArgentWalletDetectorAbi } from '@cowswap/abis'
+
+function useCheckIsSmartContract(): boolean | undefined {
+  const { provider } = useWeb3React()
+  const { account } = useWalletInfo()
+
+  const { data } = useSWR(['isSmartContract', account, provider], async () => {
+    if (!account || !provider) {
+      return false
+    }
+
+    try {
+      const code = await provider.getCode(account)
+      return code !== '0x'
+    } catch (e: any) {
+      console.debug(`checkIsSmartContractWallet: failed to check address ${account}`, e.message)
+      return false
+    }
+  })
+
+  return data
+}
+
+export function useIsSmartContractWallet(): boolean {
+  const [isSmartContractWallet, setIsSmartContractWallet] = useState<boolean>(false)
+
+  const { account } = useWalletInfo()
+
+  const isArgentWallet = useIsArgentWallet()
+  const isSmartContract = useCheckIsSmartContract()
+  const isAmbireWallet = useIsAmbireWallet()
+
+  useEffect(() => {
+    if (!account) {
+      setIsSmartContractWallet(false)
+      return
+    }
+
+    if (isAmbireWallet || isArgentWallet || isSmartContract) {
+      setIsSmartContractWallet(true)
+    }
+  }, [account, isAmbireWallet, isArgentWallet, isSmartContract])
+
+  return isSmartContractWallet
+}
+
+function useIsArgentWallet(): boolean {
```

Comment:
Moved useIsArgentWallet and useIsAmbireWallet into this file because they are used only here